src/ResourceManager/Insights/Commands.Insights/Metrics/GetAzureRmMetricDefinitionCommand.cs

```diff
@@ -72,6 +73,8 @@ protected string ProcessParameters()
         /// </summary>
         protected override void ProcessRecordInternal()
         {
+            this.WriteIdentifiedWarning("Parameter deprecation", "The DetailedOutput parameter will be deprecated in May 2018.");
```

Comment:
@gucalder I would change "May 2018" to "future breaking change release" - even though the next breaking change release _should_ be May 2018, we cannot be 100% positive that is when it is right now